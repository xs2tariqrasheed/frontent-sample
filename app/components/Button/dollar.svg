<svg xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 32 32">
    <g fill="none" fill-rule="evenodd">
        <g>
            <g>
                <g>
                    <g transform="translate(-24 -2862) translate(16 2854) translate(8 8)">
                        <circle cx="16" cy="16" r="16" fill="#02001A" fill-opacity=".16"/>
                        <text fill="#FFF" font-family="RedHatDisplay-Bold, Red Hat Display" font-size="16" font-weight="bold">
                            <tspan x="11.192" y="22">$</tspan>
                        </text>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>
