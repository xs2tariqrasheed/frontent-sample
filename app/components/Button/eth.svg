<svg xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 32 32">
    <g fill="none" fill-rule="evenodd">
        <g>
            <g>
                <g>
                    <g transform="translate(-199 -2862) translate(16 2854) translate(175) translate(8 8)">
                        <circle cx="16" cy="16" r="16" fill="#02001A" fill-opacity=".16"/>
                        <g fill="#FFF" fill-rule="nonzero">
                            <path d="M8.889 7.8l-4.445 2.6L0 7.8 4.444 0 8.89 7.8zm-4.445 4.543L0 9.6 4.444 16 8.89 9.6l-4.445 2.743z" transform="translate(12 8)"/>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>
