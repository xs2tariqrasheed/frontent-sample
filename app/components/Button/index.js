import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import ethIcon from './eth.svg';
import dollarIcon from './dollar.svg';

const Button = props => (
  <button
    type="button"
    className={classnames(
      'mb-2 btn font-body font-bold text-base text-center',
      {
        'w-auto': !props.width,
        [props.width]: props.width,
        'flex items-center p-2': props.icon,
        'text-white': props.textwhite,
        'bg-green-golf-green hover:bg-viridian active:bg-viridian': props.green,
        'bg-purple-200': props.dark,
      },
    )}
    disabled={props.isLoading}
    {...props}
  >
    {/* {props.isLoading ? ( */}
    {/*  <Spinner className="w-10 relative mx-auto block h-6 text-center" /> */}
    {/* ) : ( */}
    {props.icon && (
      <img
        className="w-8 mr-2 font-display"
        src={props.icon === 'eth' ? ethIcon : dollarIcon}
        alt=""
      />
    )}
    {props.children}
    {/* )} */}
  </button>
);

Button.propTypes = {
  width: PropTypes.string,
  textwhite: PropTypes.any,
  icon: PropTypes.string,
  dark: PropTypes.any,
  green: PropTypes.any,
  children: PropTypes.any,
  isOutline: PropTypes.any,
  isNarrow: PropTypes.any,
  isBlock: PropTypes.any,
  isLink: PropTypes.any,
  isMuted: PropTypes.any,
  isDanger: PropTypes.any,
  secondary: PropTypes.bool,
  isLoading: PropTypes.bool,
  className: PropTypes.string,
};

export default Button;
