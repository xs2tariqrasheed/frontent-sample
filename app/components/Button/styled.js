import styled, { css } from 'styled-components';
import theme from '../../styles/theme';

export const ButtonElement = styled.button`
  &:disabled {
    opacity: 0.5;
  }

  ${props =>
    props.isOutline &&
    css`
      background: white;
      color: ${theme.color.primary};
      border: 0.1rem solid ${theme.color.border};
      transition: box-shadow 200ms ease-in-out;
      &:hover {
        box-shadow: 0 1rem 4rem 0 rgba(62, 57, 107, 0.07),
          0 0.2rem 0.9rem 0 rgba(62, 57, 107, 0.06);
      }
    `}

  ${props =>
    props.isMuted &&
    css`
      background: white;
      color: ${theme.color.textMuted};
      border: 0.1rem solid ${theme.color.border};
    `}

  ${props =>
    props.isLink &&
    css`
      border: none;
      background: transparent;
      color: ${theme.color.textLight};
    `} a {
    color: inherit;
    transition: box-shadow 200ms ease-in-out;

    &:hover {
      box-shadow: 0 1rem 4rem 0 rgba(62, 57, 107, 0.07),
        0 0.2rem 0.9rem 0 rgba(62, 57, 107, 0.06);
    }
  }

  ${props =>
    props.isDanger &&
    css`
      background-color: ${theme.color.red};
    `}

  ${props =>
    props.secondary &&
    css`
      background-color: ${theme.color.lightBackground};
      color: ${theme.color.primary};
    `}
`;
