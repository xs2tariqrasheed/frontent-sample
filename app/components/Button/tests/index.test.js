import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import Button from '../index';

describe('<Button />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <Button />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <Button isOutline />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <Button isLink />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <Button isMuted />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <Button isDanger />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
