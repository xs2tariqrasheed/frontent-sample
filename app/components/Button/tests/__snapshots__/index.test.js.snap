// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render and match the snapshot 1`] = `
<button
  class="mb-2 btn font-body font-bold text-base text-center w-auto"
  type="button"
/>
`;

exports[`<Button /> should render and match the snapshot 2`] = `
<button
  class="mb-2 btn font-body font-bold text-base text-center w-auto"
  type="button"
/>
`;

exports[`<Button /> should render and match the snapshot 3`] = `
<button
  class="mb-2 btn font-body font-bold text-base text-center w-auto"
  type="button"
/>
`;

exports[`<Button /> should render and match the snapshot 4`] = `
<button
  class="mb-2 btn font-body font-bold text-base text-center w-auto"
  type="button"
/>
`;

exports[`<Button /> should render and match the snapshot 5`] = `
<button
  class="mb-2 btn font-body font-bold text-base text-center w-auto"
  type="button"
/>
`;
